monix-eval/shared/src/main/scala/monix/eval/internal/TaskStart.scala

```diff
@@ -50,10 +53,32 @@ private[eval] object TaskStart {
       // It needs its own context, its own cancelable
       val ctx2 = Task.Context(ctx.scheduler, ctx.options)
       // Starting actual execution of our newly created task;
-      Task.unsafeStartEnsureAsync(fa, ctx2, Callback.fromPromise(p))
+      val (fa2, pCtx) =
+        if (ctx.options.contextIsolationOnFork) {
+          val pCtx = Promise[Local.Context]()
+          TaskLocal.isolate(fa.map(a => { pCtx.success(Local.getContext()); a })) -> pCtx
+        } else {
+          fa -> null
+        }
+      Task.unsafeStartEnsureAsync(fa2, ctx2, Callback.fromPromise(p))
       // Signal the created fiber
       val task = Task.fromCancelablePromise(p)
-      cb.onSuccess(Fiber(task, ctx2.connection.cancel))
+      val join =
+        if (pCtx eq null) task
+        else
+          task.flatMap { a =>
+            if (pCtx ne null)
```

Comment:
Can it be not null before `flatMap` and then change?

Reply:
Well spotted - I forgot to remove that as I was moving code around 😅